client/src/api/results.js

```diff
@@ -1,148 +1,124 @@
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 3,
-      title: "Baguette",
-      category: "bread",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 4,
-      title: "Haxe",
-      category: "meat",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 5,
-      title: "Toast",
-      category: "bread",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 6,
-      title: "Kidney-Bohnen",
-      category: "beans",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 7,
-      title: "Margarine",
-      category: "oil",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 8,
-      title: "Cheddarkäse",
-      category: "cheese",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-  ];
-  return mockedResults;
+// Update offer with reserved_by user_id
+export async function reserveOffer(offerId) {
+  try {
+    const response = await fetch(`/api/offers/${offerId}`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    const result = await response.json();
+    return result;
+  } catch (error) {
+    alert(error.message);
+    return null;
+  }
 }
 
-export async function fetchOffer() {
-  const mockedResult = {
-    id: 1,
-    title: "12 frische Bananen",
-    category: "fruits",
-    tags: ["vegan", "vegetarisch", "laktosefrei", "glutenfrei"],
-    location: {
-      name: "Papa",
-      street: "Musterstraße 12",
-      zip: "50968",
-      city: "Köln",
-    },
-    date: "15.08.2020",
-    time: "15:00 - 16:00",
-  };
-  return mockedResult;
+export async function fetchReservations() {
+  try {
+    const response = await fetch("/api/offers/user/reservations", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    const result = await response.json();
+    console.log(result);
+    return result;
+  } catch (error) {
+    alert(error.message);
+    return null;
+  }
 }
 
-export async function fetchReservations(userId) {
-  const mockedResults = [
-    {
-      id: 1,
-      title: "Bananen",
-      category: "fruits",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "15:00 - 16:00",
-    },
-    {
-      id: 2,
-      title: "Joghurt",
-      category: "milk",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-    {
-      id: 3,
-      title: "Baguette",
-      category: "bread",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-  ];
-  return mockedResults;
+export async function fetchOffers() {
+  try {
+    const response = await fetch("/api/offers/user/offers", {
```

Comment:
This route is a bit uncommon.
If you want to receive offers of the current user, better change it to `/api/user/offers`.
You might face issues with other routes like `/api/offers/:offerId`
Now `user` could be an `offerId` too.

Reply:
Thank you, Leon. Changed it now.